controllers/storagecluster/external_resources_test.go

```diff
@@ -557,3 +557,43 @@ func TestExternalMonitoringResources(t *testing.T) {
 		})
 	}
 }
+
+func TestErasureCodedExternalResources(t *testing.T) {
+	cr := &api.StorageCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "ocsinit",
+		},
+		Spec: api.StorageClusterSpec{
+			ExternalStorage: api.ExternalStorageClusterSpec{
+				Enable: true,
+			},
+			Monitoring: &api.MonitoringSpec{
+				ReconcileStrategy: string(ReconcileStrategyIgnore),
+			},
+		},
+	}
+	externalResource := []ExternalResource{
+		{
+			Name: "ceph-rbd-ec",
+			Kind: "StorageClass",
+			Data: map[string]string{
+				"dataPool": generateNameForCephBlockPool(cr),
+				"pool":     generateNameForEcCephBlockPool(cr),
+			},
+		},
+		{
+			Name: "ceph-rbd",
+			Kind: "StorageClass",
+			Data: map[string]string{
+				"pool": generateNameForCephBlockPool(cr),
+			},
+		},
+	}
+	for _, extR := range externalResource {
+		t.Run(extR.Name, func(t *testing.T) {
+			sc := newCephBlockPoolStorageClassConfiguration(cr, extR.Data)
+			assert.Equal(t, sc.storageClass.Parameters["dataPool"], extR.Data["dataPool"])
+			assert.Equal(t, sc.storageClass.Parameters["pool"], extR.Data["pool"])
```

Comment:
Also test that previous fields don't go away, e,g: `clusterID`

Reply:
added.